File: frontend/src/types/project.ts
export interface Project {
  id: number;
  user_id: number;
  project_name: string;
  description: string | null;
  language: string;
  framework: string | null;
  version: string | null;
  uploaded_file_name: string | null;
  uploaded_file_size: number | null;
  uploaded_file_path: string | null;
  upload_status: string | null;
  workspace_path: string | null;
  uploaded_at: string | null;
  extraction_status: string | null;
  total_files: number | null;
  total_folders: number | null;
  extraction_time_ms: number | null;
  extracted_at: string | null;
  created_at: string;
  updated_at: string;
}

export interface ProjectListResponse {
  items: Project[];
  total: number;
  page: number;
  page_size: number;
  total_pages: number;
}

export interface ProjectCreateInput {
  project_name: string;
  description?: string;
  language: string;
  framework?: string;
  version?: string;
}

export interface ProjectUpdateInput {
  project_name?: string;
  description?: string;
  language?: string;
  framework?: string;
  version?: string;
}

File: backend/app/schemas/project.py
from datetime import datetime
from typing import Optional

from pydantic import BaseModel, Field


class ProjectCreateRequest(BaseModel):
    project_name: str = Field(..., min_length=1, max_length=255)
    description: Optional[str] = None
    language: str = Field(..., min_length=1, max_length=100)
    framework: Optional[str] = None
    version: Optional[str] = None


class ProjectUpdateRequest(BaseModel):
    project_name: Optional[str] = Field(None, min_length=1, max_length=255)
    description: Optional[str] = None
    language: Optional[str] = Field(None, min_length=1, max_length=100)
    framework: Optional[str] = None
    version: Optional[str] = None


class ProjectResponse(BaseModel):
    id: int
    user_id: int
    project_name: str
    description: Optional[str] = None
    language: str
    framework: Optional[str] = None
    version: Optional[str] = None
    uploaded_file_name: Optional[str] = None
    uploaded_file_size: Optional[int] = None
    uploaded_file_path: Optional[str] = None
    upload_status: Optional[str] = None
    workspace_path: Optional[str] = None
    uploaded_at: Optional[datetime] = None
    extraction_status: Optional[str] = None
    total_files: Optional[int] = None
    total_folders: Optional[int] = None
    extraction_time_ms: Optional[int] = None
    extracted_at: Optional[datetime] = None
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class ProjectListResponse(BaseModel):
    items: list[ProjectResponse]
    total: int
    page: int
    page_size: int
    total_pages: int
